File: demand.cpp
#include <iostream>
using namespace std;

int main() {
  int priceValue1; 
  cout << "What is the first price value?:"; 
  cin >> priceValue1;

  int priceValue2;
  cout << "What is the second price value?:"; 
  cin >> priceValue2;
  
  int priceValue3;
  cout << "What is the first demand value?:"; 
  cin >> priceValue3;
  
  int priceValue4;
  cout << "What is the second demand value?:"; 
  cin >> priceValue4;
  
  cout << endl; 
  
 //This finds the slope/b value of your question
  
  int b1 = priceValue4 - priceValue3;
  int b2 = priceValue2 - priceValue1;
  int bFin = b1/b2;
  cout << bFin;

  cout << endl;
  cout << endl;

  int a = priceValue3 - bFin * priceValue1;

  cout << a << endl;
  
  cout << endl;
  cout << endl;
  // This outputs and prints your necessary anser; the demand formula with the necessary values. 
  cout << "The demand equation is then " << "Q = " << a << " + " << bFin << "P";
  if (bFin < 0){
    cout << "The demand equation is then " << "Q = " << a << bFin << "P";
  }else{
    cout << "The demand equation is then " << "Q = " << a << " + " << bFin << "P";
  }
  
  
  return 0;
  
  

  
  
}

File: demand.py
in_1 = input("What is the first price value?: ")
in_2 = input("What is the second price value?: ")
in_3 = input("What is the first demand value?: ")
in_4 = input("What is the second demand value?: ")

#Q = a+bP

#This finds the slope/b value of your question

b_1 = float(in_4) - float(in_3)
b_2 = float(in_2) - float(in_1)
b_fin = int(b_1/b_2)
print()
print(b_fin)
print()
print(" This is your b value in the equation Q = a+bP")
print()

a = int(in_3) - int(b_fin) * int(in_1)

print(a)

print()
print("This is your a value in the equation Q = a+bP")
print()

#This outputs and prints your answer

print("The demand equation is then")
print("Q = " + str(a) + "+" + str(b_fin) + "P")




